### PR TITLE
Allow hubot to invite people to Slack

### PR DESCRIPTION
**Pull request by colmdoyle**
How do we like these apples?

### CODE REVIEW

**Comment by colmdoyle** [comment]
I can handle the merge, but I want to have the discussion about whether this is even a good idea.

**Comment by donovanh** [comment]
Interesting... so people type "hubot invite blah@blah.com" and it invites them?

File: scripts/invite.coffee [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Description:
+#   Invite people into the community!
+# Commands:
+#   hubot invite <email_address> - Lists current admins
```
Comment (IanConnolly) [inline scripts/invite.coffee]: "Lists current admins" -> "Invite a user to join the Slack group"

**Comment by colmdoyle** [comment]
@IanConnolly Yeah, spotted that and fixed.

**Comment by colmdoyle** [comment]
@donovanh exactly.

**Comment by colmdoyle** [comment]
screen shot 2015-01-21 at 11 07 12

**Comment by colmdoyle** [comment]
My only concern is people getting spammed by loads of people inviting them, or one member doing it repeatedly. 

But I'd rather assume that won't happen until it becomes an issue

**Comment by IanConnolly** [comment]
Possibly just check if a user is in the admins list?

Other than that, r+

**Comment by donovanh** [comment]
Tentatively for it. Can we monitor it for such abuse?

Not sure how visible it will be within the Slack but it does remove one step and the feeling that admins are gatekeepers.

**Comment by IanConnolly** [comment]
Log all invites to the admin channel?

**Comment by colmdoyle** [comment]
@IanConnolly Yeah, the idea is that anyone, even non-admins could use it to invite. I'll read up about posting to specific channels

@donovanh It'll be as visible as we make it. It would be up to us to push it to people. Doesn't solve our getting outside our social circles issue, but it makes it less like people need to ask nicely.

**Comment by colmdoyle** [comment]
Course, in the absence of a premium subscription, logging to #admin is going to be short lived. I could stick it in the heroku logs?

**Comment by IanConnolly** [comment]
Both would be good imo? The admins can monitor in semi-real-time to prevent any obvious abuse happening, and the heroku logs would be there to deal with any post-facto complaints?

**Comment by colmdoyle** [comment]
I can't see a way in hubot to log to specific channels, but the bot replies to successful invites

**Comment by IanConnolly** [comment]
@colmdoyle https://github.com/slackhq/hubot-slack/blob/master/src/slack.coffee#L193-L200

**Comment by colmdoyle** [comment]
@IanConnolly Is that not for replying back to the room it came from as opposed to allowing me to specify a room?

**Comment by IanConnolly** [comment]
It is, but there's a SlackClient object with a function for getting a channel object by its name? Sorry to be unclear, I was showing its use.

**Comment by EamonLeonard** [comment]
I think the spamming concern can be allayed by making sure it invites everyone at Web Summit forever, heh.

**Comment by colmdoyle** [comment]
Well, i've added some logging. I think we should proceed and we can pull back if it gets out of hand (by some definition of that)

**Comment by colmdoyle** [comment]
Any objections to merging?

**Pull request merged**